galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.resource.manager/src/main/java/dev/galasa/cicsts/resource/internal/CicsBundleImpl.java

```diff
@@ -103,31 +81,109 @@ public CicsBundleImpl(CicsResourceManagerImpl cicsResourceManager, ICicsRegion c
         this.testClass = testClass;
         this.resourceDefinitionName = name;
         this.resourceDefinitionGroup = group;
+        this.skeletonType = skeletonType;
+        logger.debug("Creating CICS Bundle Directory");      
         try {        	                          
-			this.runTemporaryUNIXPath = this.zosFileHandler.newUNIXFile(cicsRegion.getRunTemporaryUNIXDirectory().getUnixPath() + "CICSBundles" + SLASH_SYBMOL + getName() + SLASH_SYBMOL, this.cicsZosImage);
+			this.runTemporaryUNIXPath = this.zosFileHandler.newUNIXFile(cicsRegion.getRunTemporaryUNIXDirectory().getUnixPath() + "CICSBundles" + SLASH_SYBMOL , this.cicsZosImage);
 		} catch (CicstsManagerException | ZosUNIXFileException e) {
 			throw new CicsBundleResourceException("Unable to get run temporary UNIX path", e);
 		}
         // CICS bundle source already stored on file system 
         if (bundlePath == null) {
-            this.shouldDeploy = false;
+            this.needsCopying = false;
             this.resourceDefinitionBundledir = bunndledir;
         } else {
-            this.shouldDeploy = true;
+            this.needsCopying = true;
 	        if (bundlePath.endsWith(SLASH_SYBMOL)) {
 	        	this.localBundlePath = bundlePath;
 	        } else {
 	        	this.localBundlePath = bundlePath + SLASH_SYBMOL;
 	    	}
-	    	String root = new File(this.localBundlePath).getName();
-	    	this.resourceDefinitionBundledir = this.runTemporaryUNIXPath.getUnixPath() + root + SLASH_SYBMOL;
-	        if (parameters != null && !parameters.isEmpty()) {
-	        	this.parameters.putAll(parameters);
+	        this.bundleRoot = new File(this.localBundlePath).getName();
+	    	this.resourceDefinitionBundledir = this.runTemporaryUNIXPath.getUnixPath() + this.bundleRoot + SLASH_SYBMOL;
+	    	logger.info("resourceDefinitionBundledir " + resourceDefinitionBundledir);
+	        
+	    	
+	    	if (parameter != null && !parameter.isEmpty()) {
+	        	this.parameters.putAll(parameter);
 	        }
-    		this.testBundleResources = this.artifactManager.getBundleResources(this.testClass);
+    		
+			Iterator<Map.Entry<String, Object>> iterator = this.parameters.entrySet().iterator();
+			while (iterator.hasNext()) {
+				Map.Entry<String, Object> entry = iterator.next();
+				logger.info("Parameter :" + entry.getKey() +" " + entry.getValue());
+			}
+	    	this.testBundleResources = this.artifactManager.getBundleResources(this.testClass);
         }
 	}
+	
+
+
+
+	public void copyBundleFilesToZfs() throws ZosUNIXFileException, IOException {

```

Comment:
Pretty complex logic in this method. Any plans to add some unit tests to make sure it does what you expect ?